compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -54,10 +55,26 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
 
   /** Transform quoted trees while maintaining phase correctness */
   override protected def transformQuotation(body: Tree, quote: Tree)(implicit ctx: Context): Tree = {
+    val taggedTypes = new PCPCheckAndHeal.QuoteTypeTags(quote.span)(using ctx)
+    given Context =
+      if level(ctx) == 0 then contextWithQuoteTypeTags(taggedTypes)(ctx)
```

Comment:
Line 70 uses `level != 0`, is `level(ctx) == 0` the same as `level != 0`?

Reply:
Elsewhere it seems we only handle type tags `if level > 0`. Maybe add a doc to `transformQuotation`.

Reply:
I need to write `level(ctx)` or it takes the given context I'm defining recursively.

Reply:
I will simplify it